bin/PurgeBadJobs.py

```diff
@@ -0,0 +1,127 @@
+#!/miniconda-latest/bin/python3
```

Comment:
When I first wrote it I thought it would just be a Devops script using the ee2 api with a token after scanning mongodb for broken records, but then I Realized I could just use some of the ee2 libs to make it easier to get these records.  So then I just wrote these scripts and tested them on the CI env and tested that way. 

It just didn't seem like the best use of time to write tests for these, but maybe I just haven't seen the light yet. 

For a unit test, I would mock out the config,  then ee2 client, slack client, then mongo? What would I even be testing then?
For an integration test I would start up an entire ee2 instance, write some records, put them into a bad state, and then wait for the cronjob to start up and then sleep until the cronjob runs and then check the records again?

Reply:
I mean this is a script that is regularly running and making changes to a prod DB. It should definitely have tests

Reply:
For the script as is, I would just use the mongo from the docker compose and

1) create a fake deploy file with a unique DB name
2) set the deployment env
3) stick a few jobs in good states in the db
4) put a couple jobs in bad states for each state the script fixes
5) run purge()
6) pull all the jobs and check that the good jobs are unchanged and the bad jobs are fixed

Reply:
The only time I would not have tests is if the script is a one off type thing and someone is going to be monitoring it the whole way through the operation

Reply:
Alright, we can do this in a different issue once the reaper is done. https://kbase-jira.atlassian.net/jira/software/projects/DATAUP/boards/57?selectedIssue=DATAUP-558

Reply:
fair enough